### PR TITLE
Adding functional tests for Firefox Sync

### PR DESCRIPTION
**Pull request by schalkneethling**
@alexgibson r?

### CODE REVIEW

File: tests/functional/firefox-sync.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    casper.start(url, function() {
+        test.assertHttpStatus(200);
+
+        test.assertVisible('#newsletter-form', 'Newsletter form visible');
+        test.assertVisible('.cta', 'The CTA is visible');
+        test.assertVisible('#cta-android-footer', 'Download Fx for Android button visible');
+        test.assertElementCount('.secondary .features div', 3, 'List of three feature exists');
+    });
+
+    casper.waitUntilVisible('.sync-anim.on', function() {
```
Comment (alexgibson) [inline tests/functional/firefox-sync.js]: Would `waitForSelector` be a better fit here? git pull https://github.com/schalkneethling/bedrock.git casper-tests-firefox-sync

File: tests/functional/firefox-sync.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        test.assert(true, 'Sync animation visible and playing');
+    });
+
+    casper.waitUntilVisible('.sync-anim.on.complete', function() {
```
Comment (alexgibson) [inline tests/functional/firefox-sync.js]: Same here?

File: tests/functional/firefox-sync.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    casper.run(function() {
+        test.done();
+        helpers.done();
+    });
+});
+
+casper.test.begin('Firefox Sync, Firefox on Android: ' + url, 1, function suite(test) {
+
+    casper.start(url).thenEvaluate(function() {
```
Comment (alexgibson) [inline tests/functional/firefox-sync.js]: Seems like less code / dom stuff if we just set the user agent to android for this test?

File: tests/functional/firefox-sync.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    });
+});
+
+casper.test.begin('Firefox Sync, iOS: ' + url, 1, function suite(test) {
+
+    casper.start(url).thenEvaluate(function() {
+        document.documentElement.classList.add('ios');
+    }).then(function() {
+        test.assertVisible('.primary .show-ios.warning', 'iOS messaging shown');
+    }).thenEvaluate(function() {
```
Comment (alexgibson) [inline tests/functional/firefox-sync.js]: Second call to evaluate the DOM here is superfluous?

File: tests/functional/firefox-sync.js (hunk 1 above)
Comment (schalkneethling) [inline tests/functional/firefox-sync.js]: Of course :) Missed that one. thx

File: tests/functional/firefox-sync.js (hunk 3 above)
Comment (schalkneethling) [inline tests/functional/firefox-sync.js]: I initially had it like that, but from what I can see of the code in sync.js (I believe) it actually looks for those values. Also, testing revealed that just setting the UA did not trigger those instructions to be displayed.

File: tests/functional/firefox-sync.js (hunk 4 above)
Comment (schalkneethling) [inline tests/functional/firefox-sync.js]: I will change this to be as on the products tests.

File: tests/functional/firefox-sync.js (hunk 3 above)
Comment (alexgibson) [inline tests/functional/firefox-sync.js]: Oh, it's looking for the Firefox for Android UA string - did you try testing that?

Comment (schalkneethling) [inline tests/functional/firefox-sync.js]: Yup, no deal :(

Comment (alexgibson) [inline tests/functional/firefox-sync.js]: Hmm, I'll take a look

Comment (alexgibson) [inline tests/functional/firefox-sync.js]: This seems to work for me? https://gist.github.com/alexgibson/a94db414ee93a7c58013

**Comment by schalkneethling** [comment]
Alrighty, updated.

**Comment by alexgibson** [comment]
Care to squash?

**Comment by alexgibson** [comment]
r+ :sake:

**Pull request merged**